sr_robot_launch/scripts/teach_mode_node

```diff
@@ -50,33 +58,51 @@ class TeachModeManager:
         self.modes = ["TRAJECTORY_MODE", "TEACH_MODE", "POSITION_MODE", "DIRECT_PWM_MODE"]
 
     @staticmethod
-    def _create_hand_joints():
+    def _create_hand_joints(waiting_timeout: Union[int, float] = 60) -> Dict[str, list]:
+        """
+        Create a dictionary of hand joints based on the joint states.
+        This method will wait for the joint states to be published for a specified timeout in
+        increments of 10 seconds.
+        @param waiting_timeout: The timeout in seconds, default of 60 seconds.
+        @return: A dictionary of hand joints.
+        """
         hand_joint_names = {
             "rh_": [],
             "lh_": []
         }
-        try:  # pylint: disable=R1702
-            joint_states_msg = rospy.wait_for_message("/joint_states", JointState, timeout=10)
-            for prefix, joint_list in hand_joint_names.items():
-                for joint in joint_states_msg.name:
-                    if prefix in joint:
-                        if "_THJ" not in joint and "_WRJ" not in joint:
-                            if "J1" in joint:
-                                joint_list.append(joint[len(prefix):-1].lower() + "0")
-                            elif "J2" not in joint:
+        time_waited = rospy.Time.now().to_sec()
+        increment = 0
+        # Wait for the joint states to be published for a specified timeout in increments of 10 seconds.
+        while not rospy.is_shutdown() and \
+                rospy.Time.now().to_sec() - time_waited < waiting_timeout:  # pylint: disable=R1702
+            try:
+                increment += 1
+                joint_states_msg = rospy.wait_for_message("joint_states", JointState, timeout=10)
+            except rospy.ROSException:
+                rospy.logwarn(f"Waiting for joint states to be published for {increment*10} seconds...")
+            else:
+                for prefix, joint_list in hand_joint_names.items():
+                    for joint in joint_states_msg.name:
+                        if prefix in joint:
```

Comment:
you could perhaps avoid using a nested block doing this:
```suggestion
                        if prefix not in joint:
                            continue
```
and after that add all of the other stuff